doc/nvim-tree-lua.txt

```diff
@@ -1,90 +1,125 @@
+    refresh the tree
 
 |:NvimTreeFindFile|
 
-The command will change the cursor in the tree for the current bufname.
+    The command will change the cursor in the tree for the current bufname.
 
-It will also open the leafs of the tree leading to the file in the buffer
-(if you opened a file with something else than the NvimTree, like `fzf` or
-`:split`)
+    It will also open the leafs of the tree leading to the file in the buffer
+    (if you opened a file with something else than the NvimTree, like `fzf` or
+    `:split`)
 
 |:NvimTreeFindFileToggle|
 
-close the tree or change the cursor in the tree for the current bufname,
-similar to combination of |:NvimTreeToggle| and |:NvimTreeFindFile|. Takes an
-optional path argument.
+    close the tree or change the cursor in the tree for the current bufname,
+    similar to combination of |:NvimTreeToggle| and |:NvimTreeFindFile|. Takes an
+    optional path argument.
 
 |:NvimTreeClipboard|
 
-Print clipboard content for both cut and copy
+    Print clipboard content for both cut and copy
 
 |:NvimTreeResize|
 
-Resize the NvimTree window to the given size. Example: `:NvimTreeResize 50`
-resizes the window to the width of 50. If the size starts with "+" or "-" it
-adds or removes the given value to the current window width.
-Example `:NvimTreeResize -20` removes the value 20 from the current width. And
-`:NvimTreeResize +20` adds the value 20 to the current width.
+    Resize the NvimTree window to the given size. Example: `:NvimTreeResize 50`
+    resizes the window to the width of 50. If the size starts with "+" or "-" it
+    adds or removes the given value to the current window width.
+    Example `:NvimTreeResize -20` removes the value 20 from the current width. And
+    `:NvimTreeResize +20` adds the value 20 to the current width.
 
 |:NvimTreeCollapse|
 
-Collapses the nvim-tree recursively.
+    Collapses the nvim-tree recursively.
 
 |:NvimTreeCollapseKeepBuffers|
 
-Collapses the nvim-tree recursively, but keep the directories open, which are
-used in an open buffer.
+    Collapses the nvim-tree recursively, but keep the directories open, which are
+    used in an open buffer.
 
 
 ==============================================================================
-SETUP                                                *nvim-tree.setup*
+ 4. SETUP                                                   *nvim-tree-setup*
 
-To configure the tree (and make it runnable), you should call the setup
-function.
+You must run setup() function to initialise nvim-tree.
 
-Setup may only be run once; subsequent calls will result in a warning. Do not
-invoke from, say, a packer config function and then call it again later.
+setup() function takes one optional argument: configuration table. If omitted
+nvim-tree will be initialised with default configuration.
+
+Setup may only be run once; subsequent calls will result in a warning.
 
 Values may be functions. Warning: this may result in unexpected behaviour.
```

Comment:
We could be more specific: some options may be functions.

There are only a few (width) and we should ensure that they explicitly state it.

Reply:
The fact that option can be a function as well should only be documented alongside option itself. For example (with line 124 removed completely) that would be what I am thinking about:

```
    *nvim-tree.view.width*
    Width of the window, can be either a `%` string or a number representing columns.

    <reword me>If set to function it will receive arguments a b and c (or non here I guess) 
    and must return string or number.</reword me>

    Only works with `side` is `left` or `right`.
      Type: `string | number | function`, Default: `30`
```

Reply:
That is sensible and removes any ambiguity.